MoreLinq.Test/DeconstructibleEnumerableTest.cs

```diff
@@ -0,0 +1,387 @@
+#region License and Terms
+// MoreLINQ - Extensions to LINQ to Objects
+// Copyright (c) 2019 Atif Aziz. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#endregion
+
+namespace MoreLinq.Test
+{
+    using System;
+    using System.Collections.Generic;
+    using NUnit.Framework;
+
+    [TestFixture]
+    public class DeconstructibleEnumerableTest
+    {
+        [Test]
+        public void InitWithNullSource()
+        {
+            var e = Assert.Throws<ArgumentNullException>(() =>
+                new DeconstructibleEnumerable<object>(null));
+
+            Assert.That(e.ParamName, Is.EqualTo("source"));
+        }
+
+        [Test]
+        public void DeconstructDefault()
+        {
+            var e = Assert.Throws<InvalidOperationException>(() =>
+                default(DeconstructibleEnumerable<int>).Deconstruct(out _, out _));
+            Assert.That(e.Message, Contains.Substring("short"));
```

Comment:
This should be a different message than for message too short: the `DeconstructibleEnumerable<>` object is in an invalid state.

Reply:
Agree since one can't construct it with a null sequence. Fixed in a5dd44186cdc6389343928bb788a9059b692307c.